UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -34,15 +43,34 @@ public static List<ShowFilesResponseModel> ListFilesResponse(BatchSearchResponse
 
         public static List<ShowDailyFilesResponseModel> GetDailyShowFilesResponse(BatchSearchResponse searchResult)
         {
+            var distinctList = searchResult.Entries.Select(entry => new { YearWeek = entry.Attributes[5].Value, Date = entry.Attributes[0].Value, BatchPublishedDate = entry.BatchPublishedDate, BatchId = entry.BatchId }).OrderByDescending(x => x.BatchPublishedDate).ToList();
+            var groupDailyList = distinctList.GroupBy(entry => new { entry.YearWeek, entry.Date }).Select(x => x.First()).ToList();
+
+            List<BatchDetails> BatchDetailsList = new();
+            foreach (BatchDetails item in searchResult.Entries)
+            {
+                foreach (var innerItem in groupDailyList)
+                {
+                    if (innerItem.BatchId == item.BatchId)
+                    {
+                        BatchDetailsList.Add(item);
+                        break;
+                    }
+                }
+            }
+            searchResult.Count = BatchDetailsList.Count;
+            searchResult.Total = BatchDetailsList.Count;
+            searchResult.Entries = BatchDetailsList;
+         
             List<ShowDailyFilesResponseModel> showDailyFilesResponses = new ();
             List<AttributesModel> attributes = searchResult.Entries.Where(x => x.AllFilesZipSize.HasValue).Select(item => new AttributesModel
             {
                 BatchId = item.BatchId,
                 DataDate = item.Attributes.Where(x => x.Key.Equals("Data Date")).Select(x => x.Value).FirstOrDefault(),
                 WeekNumber = item.Attributes.Where(x => x.Key.Equals("Week Number")).Select(x => x.Value).FirstOrDefault(),
                 Year = item.Attributes.Where(x => x.Key.Equals("Year")).Select(x => x.Value).FirstOrDefault(),
-                YearWeek = item.Attributes.Where(x => x.Key.Equals("Year / Week")).Select(x => x.Value).FirstOrDefault(),
-                AllFilesZipSize = (long)item.AllFilesZipSize
+                YearWeek = item.Attributes.Where(x => x.Key.Replace(" ","").Equals("Year / Week".Replace(" ", ""))).Select(x => x.Value.Replace(" ", "")).FirstOrDefault(),
+                 AllFilesZipSize = (long)item.AllFilesZipSize
             }).ToList();
 
             IEnumerable<IGrouping<string, AttributesModel>> grouped = attributes.GroupBy(x => x.YearWeek);
```

Comment:
Distinct attribute model for each date can be created using BatchPublishedDate as given below 'latestDailyfiles ':

Remember to add BatchPublishedDate in AttributesModel.

```suggestion
            IEnumerable<IGrouping<string, AttributesModel>> grouped = attributes.GroupBy(x => x.YearWeek);
             foreach (IGrouping<string, AttributesModel> group in grouped)
            {
                List<AttributesModel> latestDailyfiles = group.AsEnumerable()
                                                    .OrderByDescending(x => x.BatchPublishedDate)
                                                    .GroupBy(x => Convert.ToDateTime(x.DataDate))
                                                    .Select(x => x.First())
                                                    .ToList();

                List<DailyFilesDataModel> lstDataDate = (latestDailyfiles.Select(item => new DailyFilesDataModel
                {
```

Reply:
Done as expected. Thanks for suggesting this refactoring change.